cmp/filter/FilterChooserFieldSpec.ts

```diff
@@ -14,40 +15,42 @@ import {renderToStaticMarkup} from 'react-dom/server';
 
 const {INT, NUMBER} = FieldType;
 
+export interface FilterChooserFieldSpecConfig extends BaseFilterFieldSpecConfig {
+    /**
+     * Function to produce a suitably formatted string for display to the user
+     * for any given field value.
+     */
+    valueRenderer?: FilterChooserValueRendererCb,
+    /**
+     * Function to parse user's input from a FilterChooser control into a typed data value for
+     * use in filtering comparisons.
+     */
+    valueParser?: FilterChooserValueParserCb,
+    /** Sample / representative value displayed by `FilterChooser` components to aid usability. */
+    example?: string,
+}
+
 /**
  * Filter field specification class for the typeahead `FilterChooser` component. Manages additional
  * configuration related to data values available for suggestion.
  *
  * Apps should NOT instantiate this class directly. Instead {@see FilterChooserModel.fieldSpecs}
  * for the relevant config to set these options.
+ *
+ * @internal
```

Comment:
Minor point, but I would move the `@internal` annotation to the constructor, as the class overall is (I think) part of the public API - we just don't want devs constructing these directly.